### ISSUE TITLE
Renaming "Gcloud"

### ISSUE THREAD

**Issue by blowmage** [GoogleCloudPlatform/google-cloud-ruby#789]
We have received word that we need to move away from use of "gcloud". This means we need to rename the library to "Google Cloud" and use a different top-level namespace. Consider the following code for the current library:

```ruby
require "gcloud"

gcloud = Gcloud.new "my-todo-project",
                    "/path/to/keyfile.json"
datastore = gcloud.datastore

task = datastore.find "Task", "sampleTask"
task["priority"] = 5
datastore.save task
```

Here are the 3 most likely options:

1. Rename `Gcloud` to `Google`. This is likely the easiest for users are they can search/replace their code and the formatting remains the same since `Gcloud` and `Google` have the same number of characters. There are no known naming collisions with any existing Google ruby libraries, but it is possible there may be in the future.

    ```ruby
    require "google/cloud"

    google = Google.new "my-todo-project",
                        "/path/to/keyfile.json"
    datastore = google.datastore

    task = datastore.find "Task", "sampleTask"
    task["priority"] = 5
    datastore.save task
    ```

2. Rename `Gcloud` to `Google::Cloud`. This option reinforces the brand by putting "Cloud" in the namespace. This avoids some possible naming collisions with other libraries, but changes the namespace from top-level to 2 levels deep. This may also cause confusion for non-cloud branded services such as datastore. I can see users asking, "Why is it `Google::Cloud::Datastore` instead of just `Google::Datastore`?"

    ```ruby
    require "google/cloud"

    gcloud = Google::Cloud.new "my-todo-project",
                               "/path/to/keyfile.json"
    datastore = gcloud.datastore

    task = datastore.find "Task", "sampleTask"
    task["priority"] = 5
    datastore.save task
    ```

3. Rename `Gcloud` to `GoogleCloud`. Like the current naming, this option avoid any possible naming collisions with other Google libraries by avoiding the `Google` namespace completely. I can also see users asking, "If this is Google's official client library then why isn't it using `Google` as the top-level namespace?"

    ```ruby
    require "googlecloud"

    gcloud = GoogleCloud.new "my-todo-project",
                             "/path/to/keyfile.json"
    datastore = gcloud.datastore

    task = datastore.find "Task", "sampleTask"
    task["priority"] = 5
    datastore.save task
    ```

There are many pros and cons for each option. No option is more or less correct according to ruby conventions. What should we do?

**Comment by blowmage** [comment]
I like the boldness of option 1, but the flip-side of that is that option 2 is probably a safer approach. Also, the code example for option 1 might need to be the same as option 2. I haven't worked all the details out yet, but its possible that we need to create the main gcloud object using `Google::Cloud` even if datastore uses `Google::Datastore` and not `Google::Cloud::Datastore`.

I like option 3 the least, but I think it is a viable option and should be considered.

**Comment by quartzmo** [comment]
The `require` in option 1 is: `require "google/cloud"`, is that correct?

**Comment by timanovsky** [comment]
Under rails option 2, which I like more regardless, would probably autoload without `require`, which is a plus.

**Comment by blowmage** [comment]
Assuming the new gem will be named something like `google-cloud`, what if the old `gcloud` gem only provided a dependency on `google-cloud` and some files so the old requires continue to work. So in effect the `gcloud` gem becomes a compatibility layer from the existing pre-1.0 code to migrate to the 1.0 `google-cloud` code? Does that make sense?

**Comment by blowmage** [comment]
It looks like we are going with option 2. 😄

**Comment by blowmage** [comment]
@remi @nicolasnoble @pcostell @jgeewax What do you all think of renaming `Gcloud` to `Google::Cloud`?

**Comment by jgeewax** [comment]
I'm +1 to Google::Cloud and all the code in option 2:

```sh
$ gem install google-cloud
```

(per http://guides.rubygems.org/name-your-gem/)

and then

```ruby
require "google/cloud"

gcloud = Google::Cloud.new "my-todo-project",
                           "/path/to/keyfile.json"
datastore = gcloud.datastore

task = datastore.find "Task", "sampleTask"
task["priority"] = 5
datastore.save task
```

I'm also OK calling stuff `client` or whatever... but we can use `gcloud` as a variable name if its convenient.

**Comment by jmuk** [comment]
Hi,

Sorry I'm late on the party, but some of upcoming APIs somehow chose 'google/cloud' namespace (see proto files under https://github.com/googleapis/googleapis/tree/master/google/cloud), which might cause some confusions.

**Comment by blowmage** [comment]
Yep. We are aware and are ensuring we are not conflicting with the ruby code generated from the protobuf files. Especially since gcloud-ruby will be consuming them! :)

**Comment by dmcgrath** [comment]
+1 on Option 2 (gem: google-cloud, require "google/cloud, module Google::Cloud), especially since Datastore *is* cloud branded (Its formal name is Google Cloud Datastore).

**Comment by quartzmo** [comment]
Update: PR #806 has been opened, implementing Option 2. 

PTAL (as best you can; there are many changed files) and comment on the PR!

**Comment by blowmage** [comment]
@omaray @jgeewax The repo was renamed to `google-cloud-ruby` and everything looks to be working great. The namespace change will be released tomorrow and then this issue will be closed.

**Comment by omaray** [comment]
Sweet!

**Comment by blowmage** [comment]
This was released Friday in v0.20.0.
